Templates/GraphicsGem/Template/Code/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+o3de_pal_dir(pal_source_dir ${CMAKE_CURRENT_LIST_DIR}/Source/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
+
+ly_add_target(
+    NAME ${gem_name}.Static STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        ${gem_name}_files.cmake
+        ${pal_source_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+            ${pal_source_dir}
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            Gem::Atom_RPI.Public
+            Gem::Atom_Utils.Static
+            Gem::Atom_Feature_Common
+            Gem::AtomLyIntegration_CommonFeatures.Static
+            Gem::LmbrCentral
+)
```

Comment:
This is missing the .API targets which is used to avoid exposing the internal API of the gem.
The .Static target is meant as a private `STATIC` library that is only linked with the gem.
The .API exposes the public API of the gem and should only provide an include directory INTERFACE for the "Include" directory.
Furthermore the .API target has minimal build dependencies to make it as lean as possible to use as it relates to compile time.

If anything I recommend making this Gem based off of a gem created from the [DefaultGem](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/CMakeLists.txt) template.

You can 
1. Use the `o3de.py create-gem` command to a create gem scratch gem using the default template
2. Update the gem with the Graphic files and make sure it all builds.
3. Use the `o3de.py create-from-template` command to create this Gem template again.
4. Use the `o3de.py create-gem -tn Graphics` to create a gem using this template
5. **Most importantly:** Make sure the gem instantiated from the Graphics template actually builds.

Step 5 is a requirement, as AR doesn't validate that the gem instantiated from the template builds.

A good test you can do is to add a temporary commit to the branch that instantiates the gem template and registers it with the `engine.json`, which will happen automatically if the destination folder is within engine-root.
Next, you can then run AR to make sure the Gem builds for all the AR supported platforms.

If that succeeds, can you add that information to the github description and then rebase or revert out the commit that added the temporary gem

Reply:
I did this, it compiles and the code is correct, however, there is a problem. The component does not show up in the component list, so far I haven't found the cause.

Reply:
This is done, AR passed with created gem and gem has been reverted

Reply:
Thanks for validating the create gem workflow with the new gem.